test/test_server.c

```diff
@@ -3358,6 +3358,150 @@ void test_0x3e_suppress_positive_response(void **state) {
     TEST_INT_EQUAL(UDSTpRecv(e->client_tp, buf, sizeof(buf), NULL), 0);
 }
 
+UDSErr_t fn_test_0x87_link_ctrl(UDSServer_t *srv, UDSEvent_t ev, void *arg) {
+    TEST_INT_EQUAL(ev, UDS_EVT_LinkControl);
+
+    UDSLinkCtrlArgs_t *args = arg;
+
+    switch (args->type) {
+    case 0x01:
+        TEST_INT_EQUAL(args->len, 0x01);
+        TEST_INT_EQUAL(*(uint8_t *)args->data, 0x05);
+        return UDS_PositiveResponse;
+    case 0x02:
+        TEST_INT_EQUAL(args->len, 0x03);
+        uint8_t expected_data[] = {0x02, 0x49, 0xF0};
+        TEST_MEMORY_EQUAL(args->data, expected_data, sizeof(expected_data));
+        return UDS_PositiveResponse;
+    case 0x03:
+        return UDS_PositiveResponse;
+    case 0x40:
+        return UDS_NRC_ConditionsNotCorrect;
+    }
+
+    return UDS_NRC_GeneralProgrammingFailure;
```

Comment:
nit: please use `UDS_NRC_SubFunctionNotSupported` in a default case.

I prefer to stick to the service-specific "Supported negative response codes" defined in the standard wherever possible.